docs/components/table-data.md

```diff
@@ -0,0 +1,479 @@
+
+const fetcher = async (): Promise<any> => {
+  // Fake delay
+  await new Promise((resolve) => setTimeout(resolve, 2000))
+
+  const responseData = [
+    {
+      id: 1,
+      // notice that property keys in data object correspond to each respective key in headers const
+      name: 'Leanne Graham', 
+      username: 'Bret',
+      email: 'Sincere@april.biz'
+    },
+    {
+      id: 2,
+      name: 'Ervin Howell',
+      username: 'Antonette',
+      email: 'Shanna@melissa.tv'
+    },
+    ...
+  ]
+
+  return {
+    data: responseData,
+    total: responseData.length,
+  }
+}
+
+const sortHandlerFunction = ({ key, sortColumnOrder, data }) => {
+  return data.sort((a, b) => {
+    if (key === 'username') {
+      if (sortColumnOrder === 'asc') {
+        if (a.username > b.username) {
+          return 1
+        } else if (a.username < b.username) {
+          return -1
+        }
+
+        return 0
+      } else {
+        if (a.username > b.username) {
+          return -1
+        } else if (a.username < b.username) {
+          return 1
+        }
+
+        return 0
+      }
+    }
+
+    return data
+  })
+}
+</script>
+```
+
+:::tip NOTE
+Notice that in the example above the _Username_ column is `sortable`. In this example sorting is handled on the client side, however should you want to apply server-side logic to handle sorting, you can bind your logic to the [`sort` event](#sort) and perform re-fetching of data as needed.
+:::
+
+### initialFetcherParams
+
+Fetcher function takes a single param - an object with the following properties:
+
+```ts
+interface FetcherParams {
+  pageSize?: number // the number of items to display per page
+  page?: number // the currently visible page
+  query?: string // a text string to filter table data on (defined in the searchInput prop)
+  sortColumnKey?: string // sortable column key (defined in the headers prop)
+  sortColumnOrder?: 'asc' | 'desc' // sorting order
+  offset?: string | null // the value of the offset for offset-based pagination. offset MUST be included in the fetcher params for offset-based pagination to work properly
+}
+```
+
+For ease of configuration, all properties in the `initialFetcherParams` object have default fallback values:
+
+```ts
+{ 
+  pageSize: 15,
+  page: 1,
+  query: '',
+  sortColumnKey: '',
+  sortColumnOrder: '',
+  offset: null
+}
+```
+
+### clientSort
+
+Boolean to enable client side sorting if using a fetcher that returns unpaginated data. Defaults to `false`. Use this in conjunction with [`sortHandlerFunction` prop](#sorthandlerfunction).
+
+### sortHandlerFunction
+
+Pass a custom sort handler function to handle sorting table data for specific columns.
+
+:::warning WARNING
+If [`clientSort` prop](#clientsort) is `true` but no `sortHandlerFunction` provided, KTableData will apply some default logic for client side sorting. This may produce unexpected results. We recommend **always** providing a `sortHandlerFunction` when using client side sorting.
```

Comment:
```suggestion
If [`clientSort` prop](#clientsort) is `true` but no `sortHandlerFunction` provided, KTableData will apply some default logic for client side sorting. This may produce unexpected results. We recommend **always** providing a `sortHandlerFunction` when using client-side sorting.
```